cli/internal/otel/receiver.go

```diff
@@ -261,47 +296,30 @@ func getFreePort() (int, error) {
 
 type OtelReceiverOption func(*Consumer)
 
-func WithSpanConsumer(consumerSpan ConsumeSpan) OtelReceiverOption {
-	return func(c *Consumer) {
-		c.consumeSpan = consumerSpan
-	}
-}
-
-func WithMetricConsumer(consumerMetric ConsumeMetric) OtelReceiverOption {
-	return func(c *Consumer) {
-		c.consumeMetric = consumerMetric
-	}
-}
-
 func WithMetricsFilename(filename string) OtelReceiverOption {
 	return func(c *Consumer) {
 		c.metricsFilename = filename
 	}
 }
 
 func StartOtelReceiver(ctx context.Context, opts ...OtelReceiverOption) (*OtelReceiver, error) {
-	c := Consumer{}
-	for _, opt := range opts {
-		opt(&c)
+	c := Consumer{
+		consumeSpan: newDefaultSpanConsumer(),
+		consumeLogs: newDefaultLogsConsumer(),
 	}
 
-	if c.consumeSpan == nil {
-		c.consumeSpan = newDefaultSpanConsumer()
+	for _, opt := range opts {
+		opt(&c)
 	}
 
-	if c.consumeMetric == nil {
-		if c.metricsFilename == "" {
```

Comment:
`metricsFilename` is never empty. We always call `StartOtelReceiver` with a non empty metrics file